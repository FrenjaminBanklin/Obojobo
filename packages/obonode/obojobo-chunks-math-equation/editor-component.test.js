import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import MathEquation from './editor-component'

jest.mock('obojobo-document-engine/src/scripts/common/util/modal-util')
jest.mock('obojobo-document-engine/src/scripts/oboeditor/components/node/editor-component', () => (
	props => <div>{props.children}</div>
))

describe('MathEquation Editor Node', () => {
	test('renders with no latex', () => {
		const component = renderer.create(
			<MathEquation
				node={{
					data: {
						get: () => {
							return { latex: null }
						}
					}
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('MathEquation component with error', () => {
		const component = renderer.create(
			<MathEquation
				node={{
					data: {
						get: () => {
							return { latex: 'x_0_0' }
						}
					}
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('MathEquation component with label', () => {
		const component = renderer.create(
			<MathEquation
				node={{
					data: {
						get: () => {
							return { latex: '1', label: '', size: 1 }
						}
					}
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('MathEquation component edits properties', () => {
		const editor = {
			setNodeByKey: jest.fn()
		}

		const component = mount(
			<MathEquation
				node={{
					data: {
						get: () => {
							return { latex: '1', label: '1.1' }
						}
					}
				}}
				isFocused={true}
				isSelected={true}
				editor={editor}
			/>
		)
		component
			.find('input')
			.at(0)
			.simulate('click', { stopPropagation: jest.fn() })
		component
			.find('input')
			.at(1)
			.simulate('click', { stopPropagation: jest.fn() })
		component
			.find('input')
			.at(2)
			.simulate('click', { stopPropagation: jest.fn() })
		component
			.find('input')
			.at(3)
			.simulate('click', { stopPropagation: jest.fn() })
		component
			.find('input')
			.at(1)
			.simulate('change', { stopPropagation: jest.fn(), target: { value: 'mockValue'} })

		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('changeProperties edits properties', () => {
		const editor = {
			setNodeByKey: jest.fn()
		}

		const component = mount(
			<MathEquation
				node={{
					data: {
						get: () => {
							return { latex: '1', label: '', size: 1 }
						}
					}
				}}
				isFocused={true}
				isSelected={true}
				editor={editor}
			/>
		)

		component.instance().changeProperties({ mockProperties: 'mock value' })

		expect(editor.setNodeByKey).toHaveBeenCalled()
	})
})
