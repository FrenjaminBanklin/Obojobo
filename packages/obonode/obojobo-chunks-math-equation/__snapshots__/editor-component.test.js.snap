// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MathEquation Editor Node MathEquation component edits properties 1`] = `"<div><div class=\\"component obojobo-draft--chunks--math-equation pad align-center\\" contenteditable=\\"false\\"><div class=\\"non-editable-chunk\\"><div class=\\"katex-container\\" style=\\"font-size: 20em;\\">mock-katex-render-for-1-with-options-{\\"displayMode\\":true}</div><div class=\\"equation-label\\">mockValue</div></div><div class=\\"attributes-box  is-open\\" contenteditable=\\"false\\"><div class=\\"box-border\\"><div class=\\"attributes-list\\"><div class=\\"attribute\\"><label for=\\"math-equation-latex\\">Latex:</label><input id=\\"math-equation-latex\\" value=\\"1\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-label\\">Optional Label:</label><input id=\\"math-equation-label\\" value=\\"mockValue\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-alt\\">Alt Text:</label><input id=\\"math-equation-alt\\" value=\\"\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-size\\">Size:</label><input id=\\"math-equation-size\\" type=\\"number\\" step=\\"0.1\\" max=\\"20\\" min=\\"0.1\\" value=\\"20\\"></div><div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\">Done</button></div></div></div></div></div></div></div>"`;

exports[`MathEquation Editor Node MathEquation component restricts minimum size to 0.1 1`] = `"<div><div class=\\"component obojobo-draft--chunks--math-equation pad align-center\\" contenteditable=\\"false\\"><div class=\\"non-editable-chunk\\"><div class=\\"katex-container\\" style=\\"font-size: 0.1em;\\">mock-katex-render-for-1-with-options-{\\"displayMode\\":true}</div><div class=\\"equation-label\\">1.1</div></div><div class=\\"attributes-box  is-open\\" contenteditable=\\"false\\"><div class=\\"box-border\\"><div class=\\"attributes-list\\"><div class=\\"attribute\\"><label for=\\"math-equation-latex\\">Latex:</label><input id=\\"math-equation-latex\\" value=\\"1\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-label\\">Optional Label:</label><input id=\\"math-equation-label\\" value=\\"1.1\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-alt\\">Alt Text:</label><input id=\\"math-equation-alt\\" value=\\"\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-size\\">Size:</label><input id=\\"math-equation-size\\" type=\\"number\\" step=\\"0.1\\" max=\\"20\\" min=\\"0.1\\" value=\\"0.1\\"></div><div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\">Done</button></div></div></div></div></div></div></div>"`;

exports[`MathEquation Editor Node MathEquation component sets blank values to 1 1`] = `"<div><div class=\\"component obojobo-draft--chunks--math-equation pad align-center\\" contenteditable=\\"false\\"><div class=\\"non-editable-chunk\\"><div class=\\"katex-container\\" style=\\"font-size: 1em;\\">mock-katex-render-for-1-with-options-{\\"displayMode\\":true}</div><div class=\\"equation-label\\">1.1</div></div><div class=\\"attributes-box  is-open\\" contenteditable=\\"false\\"><div class=\\"box-border\\"><div class=\\"attributes-list\\"><div class=\\"attribute\\"><label for=\\"math-equation-latex\\">Latex:</label><input id=\\"math-equation-latex\\" value=\\"1\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-label\\">Optional Label:</label><input id=\\"math-equation-label\\" value=\\"1.1\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-alt\\">Alt Text:</label><input id=\\"math-equation-alt\\" value=\\"\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-size\\">Size:</label><input id=\\"math-equation-size\\" type=\\"number\\" step=\\"0.1\\" max=\\"20\\" min=\\"0.1\\" value=\\"1\\"></div><div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\">Done</button></div></div></div></div></div></div></div>"`;

exports[`MathEquation Editor Node MathEquation component with error 1`] = `
<div>
  <div
    className="component obojobo-draft--chunks--math-equation pad align-center"
    contentEditable={false}
    onClick={[Function]}
  >
    <div
      className="katex-container katex-error"
    >
      <p>
        Mock katex render error
      </p>
    </div>
  </div>
</div>
`;

exports[`MathEquation Editor Node MathEquation component with label 1`] = `
<div>
  <div
    className="component obojobo-draft--chunks--math-equation pad align-center"
    contentEditable={false}
    onClick={[Function]}
  >
    <div
      className="non-editable-chunk"
    >
      <div
        className="katex-container"
        dangerouslySetInnerHTML={
          Object {
            "__html": "mock-katex-render-for-1-with-options-{\\"displayMode\\":true}",
          }
        }
        style={
          Object {
            "fontSize": "1em",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`MathEquation Editor Node renders with no latex 1`] = `
<div>
  <div
    className="component obojobo-draft--chunks--math-equation pad align-center"
    contentEditable={false}
    onClick={[Function]}
  >
    <div
      className="non-editable-chunk"
    >
      <div
        className="katex-display"
      >
        <span
          className="no-equation"
        >
          No Equation (Click to edit)
        </span>
      </div>
    </div>
  </div>
</div>
`;
